frontend/src/metabase/styled-components/theme/css-variables.ts

```diff
@@ -37,70 +51,21 @@ export function getMetabaseCssVariables(theme: MantineTheme) {
  * You don't need to add new colors from `colors.module.css` here since they'll already
  * be available globally at :root
  **/
-function getDesignSystemCssVariables(theme: MantineTheme) {
+function getSdkDesignSystemCssVariables(theme: MantineTheme) {
   return css`
     --mb-default-font-family: "${theme.fontFamily}";
 
-    --mb-color-bg-light: ${theme.fn.themeColor("bg-light")};
-    --mb-color-bg-dark: ${theme.fn.themeColor("bg-dark")};
-    --mb-color-brand: ${theme.fn.themeColor("brand")};
-
-    --mb-color-brand-light: color-mix(in srgb, var(--mb-color-brand), #fff 80%);
-    --mb-color-brand-lighter: color-mix(
-      in srgb,
-      var(--mb-color-brand),
-      #fff 90%
-    );
-    --mb-color-brand-alpha-04: color-mix(
-      in srgb,
-      var(--mb-color-brand) 4%,
-      transparent
-    );
-    --mb-color-brand-alpha-88: color-mix(
-      in srgb,
-      var(--mb-color-brand) 88%,
-      transparent
-    );
-    --mb-color-border-alpha-30: color-mix(
-      in srgb,
-      var(--mb-color-border) 30%,
-      transparent
-    );
-    --mb-color-text-white-alpha-85: color-mix(
-      in srgb,
-      var(--mb-color-text-white) 85%,
-      transparent
-    );
-    --mb-color-bg-black-alpha-60: color-mix(
-      in srgb,
-      var(--mb-color-bg-black) 60%,
-      transparent
-    );
-    --mb-color-bg-white-alpha-15: color-mix(
-      in srgb,
-      var(--mb-color-bg-white) 15%,
-      transparent
-    );
-
     /* Semantic colors */
-    --mb-color-focus: ${theme.fn.themeColor("focus")};
-    --mb-color-bg-white: ${theme.fn.themeColor("bg-white")};
-    --mb-color-bg-black: ${theme.fn.themeColor("bg-black")};
-    --mb-color-shadow: ${theme.fn.themeColor("shadow")};
-    --mb-color-border: ${theme.fn.themeColor("border")};
-    --mb-color-text-dark: ${theme.fn.themeColor("text-dark")};
-    --mb-color-text-medium: ${theme.fn.themeColor("text-medium")};
-    --mb-color-text-light: ${theme.fn.themeColor("text-light")};
-    --mb-color-danger: ${theme.fn.themeColor("danger")};
-    --mb-color-error: ${theme.fn.themeColor("error")};
-    --mb-color-filter: ${theme.fn.themeColor("filter")};
-    --mb-color-bg-error: ${color("bg-error")};
-    --mb-color-bg-medium: ${theme.fn.themeColor("bg-medium")};
-    --mb-color-bg-night: ${color("bg-night")};
-    --mb-color-text-white: ${theme.fn.themeColor("text-white")};
-    --mb-color-success: ${theme.fn.themeColor("success")};
-    --mb-color-summarize: ${theme.fn.themeColor("summarize")};
-    --mb-color-warning: ${theme.fn.themeColor("warning")};
+    /* Dynamic colors from SDK */
+    ${Object.entries(SDK_TO_MAIN_APP_COLORS_MAPPING).flatMap(
```

Comment:
remove all the colors and only define necessary custom properties. Since these are the only dynamic values, we don't need to override more than necessary. e.g. we don't support overriding `success` yet, so defining `--mb-color-success: ${theme.fn.themeColor("success")};` does nothing.